components/script/dom/document.rs

```diff
@@ -1206,12 +1206,44 @@ impl DocumentMethods for Document {
 
     // https://dom.spec.whatwg.org/#dom-document-characterset
     fn CharacterSet(&self) -> DOMString {
-        self.encoding_name.borrow().clone()
+        match (*(self.encoding_name.borrow())).as_ref() {
+            "utf-8"         =>  "UTF-8",
+            "ibm866"        =>  "IBM866",
+            "iso-8859-2"    =>  "ISO-8859-2",
+            "iso-8859-3"    =>  "ISO-8859-3",
+            "iso-8859-4"    =>  "ISO-8859-4",
+            "iso-8859-5"    =>  "ISO-8859-5",
+            "iso-8859-6"    =>  "ISO-8859-6",
+            "iso-8859-7"    =>  "ISO-8859-7",
+            "iso-8859-8"    =>  "ISO-8859-8",
+            "iso-8859-8-i"  =>  "ISO-8859-8-I",
+            "iso-8859-10"   =>  "ISO-8859-10",
+            "iso-8859-13"   =>  "ISO-8859-13",
+            "iso-8859-14"   =>  "ISO-8859-14",
+            "iso-8859-15"   =>  "ISO-8859-15",
+            "iso-8859-16"   =>  "ISO-8859-16",
+            "koi8-r"        =>  "KOI8-R",
+            "koi8-u"        =>  "KOI8-U",
+            "gbk"           =>  "GBK",
+            "big5"          =>  "Big5",
+            "euc-jp"        =>  "EUC-JP",
+            "iso-2022-jp"   =>  "ISO-2022-JP",
+            "shift_jis"     =>  "Shift_JIS",
+            "euc-kr"        =>  "EUC-KR",
+            "utf-16be"      =>  "UTF-16BE",
+            "utf-16le"      =>  "UTF-16LE",
+            name            =>  name
+        }.to_owned()
+    }
+
+    //https://dom.spec.whatwg.org/#dom-document-charset
+    fn Charset(&self) -> DOMString {
+        self.CharacterSet()
```

Comment:
@jdm I did case-converted in the `pub fn set_encoding_name`. Does that look good?